addons/web/static/src/js/views/form_widgets.js

```diff
@@ -455,7 +455,7 @@ var FieldCharDomain = common.AbstractField.extend(common.ReinitializeFieldMixin,
 
         if (this.get('value')) {
             var model = this.options.model || this.field_manager.get_field_value(this.options.model_field);
-            var domain = pyeval.eval('domain', this.get('value'));
+            try{var domain = pyeval.eval('domain', this.get('value'));} catch(e){this.do_warn(_t('Error : Bad domain'), _t('The domain is wrong.')); return;}
```

Comment:
@simongoffin could it be `Error: Bad domain` in place of `Error : Bad domain`? Looks a bit cleaner to me.